src/main/java/com/googleintern/wfm/ruleengine/action/CsvParser.java

```diff
@@ -0,0 +1,135 @@
+package src.main.java.com.googleintern.wfm.ruleengine.action;
+
+import com.opencsv.CSVReader;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.exceptions.CsvException;
+import src.main.java.com.googleintern.wfm.ruleengine.model.PoolAssignmentModel;
+import src.main.java.com.googleintern.wfm.ruleengine.model.UserPoolAssignmentModel;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * CsvParser class is used to retrieve information from the input csv file ans store these data as
+ * instances of UserPoolAssignmentModel.
+ *
+ * <p>enum variable Header represents the relationship between column position and the data stored
+ * inside. Pattern variable NUMBER_PATTERN finds numbers from strings. Pattern variable
+ * ROLESKILL_PATTERN finds skill_id elements from strings. Pattern variable PERMISSION_PATTERN finds
+ * pool_permission elements from strings.
+ */
+public class CsvParser {
+  enum Header {
+    USER_ID(0),
+    ROLE_IDS(1),
+    SKILL_IDS(2),
+    ROLESKILL_IDS(3),
+    WORKFORCE_ID(4),
+    WORKGROUP_ID(5);
+
+    final int column;
+
+    Header(int column) {
+      this.column = column;
+    }
+  }
+
+  private static final Pattern NUMBER_PATTERN = Pattern.compile("(\\d+)");
+  private static final Pattern ROLESKILL_PATTERN = Pattern.compile("\"skill_id\":\"(\\d+)\"");
+  private static final Pattern PERMISSION_PATTERN =
+      Pattern.compile("\"cases_pool_id\":\"(\\d+)\",\"permission_set_id\":\"(\\d+)\"");
+
+  /**
+   * Read all data using csvFilePath. Parse useful information out and store it in *
+   * userPoolAssignmentList.
+   *
+   * @param csvFilePath: represents the path to the csv file, including the file name.
+   * @return
+   * @throws IOException
+   * @throws CsvException
+   */
+  public static List<UserPoolAssignmentModel> readFromCSVFile(String csvFilePath)
+      throws IOException, CsvException {
+    // Read all data from input csv file located at given path
+    Reader reader = Files.newBufferedReader(Paths.get(csvFilePath));
+    CSVReader csvReader = new CSVReaderBuilder(reader).withSkipLines(1).build();
+    List<String[]> userRecords = csvReader.readAll();
+
+    // List variable userPoolAssignmentList stores the results reading from the input csv file
+    List<UserPoolAssignmentModel> userPoolAssignmentList = new ArrayList<>();
```

Comment:
Declare your variable with the specific list type and return with interfaces. 

ie: since you are building userPoolAssignmentList you should use specific list
```
ImmutableList.Builder<UserPoolAssignmentModel> userPoolAssignmentList = new ImmutableList.Builder<>();
...
userPoolAssignmentList.add()...
return userPoolAssignmentList.build();
```